spec/integration/action/format_config_spec.rb

```diff
@@ -0,0 +1,194 @@
+# frozen_string_literal: true
+
+require "json"
+require "rack/test"
+
+RSpec.describe "App action / Format config", :app_integration do
+  include Rack::Test::Methods
+
+  let(:app) { Hanami.app }
+
+  around do |example|
+    with_tmp_directory(Dir.mktmpdir, &example)
+  end
+
+  it "adds a body parser middleware for the accepted formats from the action config" do
+    write "config/app.rb", <<~RUBY
+      require "hanami"
+
+      module TestApp
+        class App < Hanami::App
+          config.logger.stream = StringIO.new
+
+          config.actions.format :json
+        end
+      end
+    RUBY
+
+    write "config/routes.rb", <<~RUBY
+      module TestApp
+        class Routes < Hanami::Routes
+          post "/users", to: "users.create"
+        end
+      end
+    RUBY
+
+    write "app/action.rb", <<~RUBY
+      # auto_register: false
+
+      module TestApp
+        class Action < Hanami::Action
+        end
+      end
+    RUBY
+
+    write "app/actions/users/create.rb", <<~RUBY
+      module TestApp
+        module Actions
+          module Users
+            class Create < TestApp::Action
+              def handle(req, res)
+                res.body = req.params[:users].join("-")
+              end
+            end
+          end
+        end
+      end
+    RUBY
+
+    require "hanami/boot"
+
+    post(
+      "/users",
+      JSON.generate("users" => %w[jane john jade joe]),
+      "CONTENT_TYPE" => "application/json"
+    )
+
+    expect(last_response).to be_successful
+    expect(last_response.body).to eql("jane-john-jade-joe")
+  end
+
+  specify "adds a body parser middleware configured to parse any custom content type for the accepted formats" do
+    write "config/app.rb", <<~RUBY
+      require "hanami"
+
+      module TestApp
+        class App < Hanami::App
+          config.logger.stream = StringIO.new
+
+          config.actions.formats.add :json, ["application/json+scim", "application/json"]
```

Comment:
@timriley Nitpick, why don't:

```ruby
config.actions.formats.add(json: ["application/json+scim", "application/json"])
```

Reply:
@jodosha so my thinking is that `add(format_name: ["mime_types", here"])` implies that _multiple_ formats and mime types can be added in one go, because it looks like a hash, and hashes are unbounded. But that's not how this method actually works, so I thought I'd make everything clearer by just acknowledging these are two distinct parameters.

Reply:
If we feel like it I'm sure we could update it to accept a single open-ended hash in a later enhancement!